src/lib.rs

```diff
@@ -928,6 +934,53 @@ impl Config {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct Version {
+    major: u32,
+    minor: u32,
+}
+
+impl Version {
+    fn new(major: u32, minor: u32) -> Self {
+        Self { major, minor }
+    }
+
+    fn parse(s: &str) -> Option<Self> {
+        // As of 3.22, the format of the version output is "cmake version <major>.<minor>.<patch>".
+        // ```
+        // $ cmake --version
+        // cmake version 3.22.2
+        //
+        // CMake suite maintained and supported by Kitware (kitware.com/cmake).
+        // ```
+        let version = s.lines().find(|s| s.starts_with("cmake version "))?;
+        let version = &version["cmake version ".len()..]; // strip prefix
```

Comment:
If we can assume it's always on the first line, you can simplify this to:
```suggestion
        let version = s.lines().next()?.strip_prefix("cmake version ")?
```

Reply:
Package managers tend to add extra strings to the version output, so I've adopted a redundant one, but the version output of binaries installed in Ubuntu (apt), macOS (homebrew), and Windows (scoop) were all the same, so I think it is safe to simplify.

As for strip_prefix, I used a method that is compatible with older compilers because I didn't know the MSRV of this crate, but given that master is the 2021 edition that requires the newer rustc, it seems I no longer need to care about that.